jupyterhub/spawner.py

```diff
@@ -306,6 +307,53 @@ def _default_access_scopes(self):
             f"access:servers!user={self.user.name}",
         ]
 
+    group_overrides = Union(
+        [Callable(), Dict()],
+        help="""
+        Override specific traitlets based on group membership of the user.
+
+        This can be a dict, or a callable that returns a dict. The keys of the dict
+        are *only* used for lexicographical sorting, to guarantee consistent
+        ordering of the overrides. If it is a callable, it may be async, and will
+        be passed one parameter - the spawner instance. It should return a dictionary.
+
+        The value of the dict is a dict, with the following keys:
+
+        - *groups* - If the user belongs to *any* of these groups, these overrides are
+          applied to their server before spawning.
+        - *spawner_override* - a dictionary with overrides to apply to the Spawner
+          settings. Each value can be either the final value to change or a callable that
+          take the `Spawner` instance as parameter and returns the final value.
+          If the traitlet being overriden is a *dictionary*, the dictionary
+          will be *recursively updated*, rather than overriden. If you want to
+          remove a key, set its value to `None`
+
+        The following example config will:
+
+        1. Add the environment variable "AM_I_USER" to everyone in the "user" group
+        2. Add the environment variable "AM_I_ADMIN" to everyone in the "admin" group.
+           If a user is part of both "user" and "admin", they will get *both* these env
+           vars, due to the dictionary merging functionality.
+        3. Add a higher memory limit for everyone in the "admin" group.
+
+        c.Spawner.group_overrides = {
+            "01-admin-env-add": {
+                "groups": ["admin"],
```

Comment:
The example leads me to gures there may be user and admin groups by default to represent all users and all admins, but there isn't right?

Maybe student and instructor or similar could be used instead to not cause someone to incorrectly assume user/admin is available?

Reply:
I don't think it matters much, I've changed the names.